pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2,7 +2,17 @@
 
 with import ./lib.nix { inherit pkgs; };
 
-self: super: {
+self: super:
+
+let
+  # apply an override to a package if it’s build with GHC <8.0
+  ifLT8 = override: drv:
```

Comment:
Yeah, but it's confusing for readers of the file to define a "local helper function" some 1000 lines of code apart from the place where it's actually used.